src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/CoreEventSource.cs

```diff
@@ -9,7 +10,29 @@ internal sealed class CoreEventSource : EventSource
     {
         public static readonly CoreEventSource Log = new CoreEventSource();
 
-        private readonly ApplicationNameProvider nameProvider = new ApplicationNameProvider();
+        private readonly string applicationName;
+
+        private CoreEventSource()
+        {
+            //// We want to add application name to all events BUT
+            //// It is prohibited by EventSource rules to have more parameters in WriteEvent that in event source method
+            //// Parameter will be available in payload but in the next versions EventSource may 
+            //// start validating that number of parameters match
+            //// It is not allowed to call additional methods, only WriteEvent
+
+            try
```

Comment:
I get that you want to set the applicationName just once in the constructor, but can't you still use ApplicationNameProvider instead of repeating the code?